### PR TITLE
Added user login

### PR DESCRIPTION
**Pull request by RobertCrabtreeKainos**

### CODE REVIEW

File: src/app.ts [hunk 1]
```diff
@@ -1,6 +1,8 @@
 import express from 'express';
 import nunjucks from 'nunjucks';
 import path from 'path';
+import session = require("express-session");
```
Comment (TomBillingtonUK) [inline src/app.ts]: `import expressSession from 'express-session';`

File: src/controller/authController.ts [hunk 2]
```diff
@@ -0,0 +1,24 @@
+import {Request, Response, Application, response} from "express";
+import { Login } from "../model/auth";
+const authService = require("../service/authService");
+
+module.exports = function(app: Application)
```
Comment (TomBillingtonUK) [inline src/controller/authController.ts]: To make this a bit more user-friendly, could you import/export a more typescript way?

Replace module.exports
```
export const authController => (app: Application) {
     ... code here
}
```
In the app.ts file you can import the controller like this:
`import { authController } from 'path'`

You can then setup the routes by doing:
`authController(app)`

File: src/middleware/auth.ts [hunk 3]
```diff
@@ -0,0 +1,11 @@
+import { NextFunction, Request, Response } from "express";
+
+module.exports = function (req: Request, res:Response, next:Function)
```
Comment (TomBillingtonUK) [inline src/middleware/auth.ts]: You could export this the same was as the example above

File: src/service/authService.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Login } from "../model/auth";
+const axios = require("axios");
+
+module.exports.login = async function (login: Login): Promise<void>
```
Comment (TomBillingtonUK) [inline src/service/authService.ts]: Export as per the example above

File: src/service/authService.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+import { Login } from "../model/auth";
+const axios = require("axios");
```
Comment (TomBillingtonUK) [inline src/service/authService.ts]: import axios from 'axios'

File: src/controller/authController.ts [hunk 6]
```diff
@@ -0,0 +1,24 @@
+import {Request, Response, Application, response} from "express";
+import { Login } from "../model/auth";
+const authService = require("../service/authService");
```
Comment (TomBillingtonUK) [inline src/controller/authController.ts]: Should be an import and not a require. 

Also check to see if the imports should be import or import type

File: src/middleware/auth.ts (hunk 3 above)
Comment (TomBillingtonUK) [inline src/middleware/auth.ts]: You resolved the previous comment and didn't make the change

**Review by TomBillingtonUK — APPROVED** [review]

**Review by sammy-perry-kainos — APPROVED** [review]

**Pull request merged**